webapp/src/components/expanded_view/controls_button.tsx

```diff
@@ -0,0 +1,131 @@
+import React from 'react';
+import {OverlayTrigger, Tooltip} from 'react-bootstrap';
+import styled, {css} from 'styled-components';
+
+import CompassIcon from 'src/components/icons/compassIcon';
+import Shortcut from 'src/components/shortcut';
+
+export type Props = {
+    id: string,
+    icon: React.ReactNode,
+    bgColor: string,
+    text?: string,
+    tooltipText: string,
+    tooltipSubtext?: string,
+    onToggle?: () => void,
+    unavailable?: boolean,
+    disabled?: boolean,
+    iconFill?: string,
+    shortcut?: string,
+    margin?: string,
+}
+
+export default function ControlsButton(props: Props) {
+    return (
+        <OverlayTrigger
+            key={props.id}
+            placement='top'
+            overlay={
+                props.disabled ||
+                <Tooltip id={`tooltip-${props.id}`}>
+                    <div>{props.tooltipText}</div>
+                    {props.tooltipSubtext &&
+                    <TooltipSubtext>
+                        {props.tooltipSubtext}
+                    </TooltipSubtext>
+                    }
+                    { props.shortcut &&
+                    <Shortcut shortcut={props.shortcut}/>
+                    }
+                </Tooltip>
+            }
+        >
+            <ButtonContainer
+                id={props.id}
+                margin={props.margin}
+            >
+                <Button
+                    className='button-center-controls'
+                    // eslint-disable-next-line no-undefined
+                    onClick={props.disabled ? undefined : props.onToggle}
+                    bgColor={props.bgColor}
+                    isDisabled={props.disabled}
+                    isUnavailable={props.unavailable}
+                    disabled={props.disabled}
+                >
+                    <ButtonIcon>
+                        {props.icon}
+                        { props.unavailable &&
+                        <UnavailableIcon>
+                            <CompassIcon icon='close-circle'/>
+                        </UnavailableIcon>
+                        }
+                    </ButtonIcon>
+                </Button>
+                { props.text &&
+                <ButtonText>{props.text}</ButtonText>
+                }
+            </ButtonContainer>
+        </OverlayTrigger>
+    );
+}
+
+const ButtonContainer = styled.div<{margin?: string}>`
+   display: flex;
+   flex-direction: column;
+   align-items: center;
+   justify-content: center;
+   margin: ${({margin}) => margin || '0 16px'};
+`;
+
+const Button = styled.button<{bgColor: string, isDisabled?: boolean, isUnavailable?: boolean}>`
+  &&& {
```

Comment:
curious why we need the extra specificity (&&&) here?

Reply:
Some definitions when implementing `WidgetButton` were definitely conflicting with some other in-product (coming from webapp that is) stylings of buttons and likely tried to prevent the same when porting the same to the expanded view although it doesn't seem to be needed there. I removed them, thanks :+1: